extensions/azdata/src/extension.ts

```diff
@@ -119,40 +119,66 @@ export async function activate(context: vscode.ExtensionContext): Promise<azdata
 				sql: {
 					mi: {
 						delete: async (name: string) => {
-							await throwIfNoAzdataOrEulaNotAccepted();
+							throwIfNoAzdataOrEulaNotAccepted();
 							return localAzdata!.arc.sql.mi.delete(name);
 						},
 						list: async () => {
-							await throwIfNoAzdataOrEulaNotAccepted();
+							throwIfNoAzdataOrEulaNotAccepted();
 							return localAzdata!.arc.sql.mi.list();
 						},
 						show: async (name: string) => {
-							await throwIfNoAzdataOrEulaNotAccepted();
+							throwIfNoAzdataOrEulaNotAccepted();
 							return localAzdata!.arc.sql.mi.show(name);
 						}
 					}
 				}
 			},
+			getPath: () => {
+				throwIfNoAzdata();
+				return localAzdata!.getPath();
+			},
 			login: async (endpoint: string, username: string, password: string) => {
-				await throwIfNoAzdataOrEulaNotAccepted();
+				throwIfNoAzdataOrEulaNotAccepted();
 				return localAzdata!.login(endpoint, username, password);
 			},
+			getSemVersion: () => {
+				throwIfNoAzdata();
+				return localAzdata!.getSemVersion();
+			},
 			version: async () => {
-				await throwIfNoAzdataOrEulaNotAccepted();
+				throwIfNoAzdata();
 				return localAzdata!.version();
 			}
 		}
 	};
 }
 
-async function throwIfNoAzdataOrEulaNotAccepted(): Promise<void> {
-	if (!localAzdata) {
-		Logger.log(loc.noAzdata);
-		throw new Error(loc.noAzdata);
+
+export class NoAzdataEulaError extends Error {
```

Comment:
Why did you make these error classes anyways? You're not doing anything special for them so just doing `throw new Error(loc.eulaNotAccepted)` would have been simpler

Reply:
I intended to use c them but then could not figure out how to expose them through azdata-ext.d.ts so went c another way for the time being

Reply:
reverted